api/src/main/java/com/okta/idx/sdk/api/wrapper/AuthenticationWrapper.java

```diff
@@ -0,0 +1,264 @@
+            logger.error("Something went wrong! {}, {}", e, errors);
+            authenticationResponse.setErrors(errors);
+            authenticationResponse.setAuthenticationStatus(AuthenticationStatus.FAILURE);
+        } catch (IllegalArgumentException e) {
+            logger.error("Exception occurred", e);
+            authenticationResponse.setAuthenticationStatus(AuthenticationStatus.FAILURE);
+        }
+
+        return authenticationResponse;
+    }
+
+    public static AuthenticationResponse changePassword(IDXClient client, IDXClientContext idxClientContext, ChangePasswordOptions changePasswordOptions) {
+        AuthenticationResponse authenticationResponse = new AuthenticationResponse();
+        TokenResponse tokenResponse;
+
+        try {
+            // re-enter flow with context
+            IDXResponse introspectResponse = client.introspect(idxClientContext);
+
+            // verify if password expired
+            if (!isRemediationRequireCredentials(RemediationType.REENROLL_AUTHENTICATOR, introspectResponse)) {
+                logger.error("Unexpected remediation {}", RemediationType.REENROLL_AUTHENTICATOR);
+            } else {
+
+                RemediationOption[] remediationOptions = introspectResponse.remediation().remediationOptions();
+                printRemediationOptions(remediationOptions);
+
+                RemediationOption remediationOption = extractRemediationOption(remediationOptions, RemediationType.REENROLL_AUTHENTICATOR);
+
+                // set new password
+                Credentials credentials = new Credentials();
+                credentials.setPasscode(changePasswordOptions.getNewPassword().toCharArray());
+
+                // build answer password authenticator challenge request
+                AnswerChallengeRequest passwordAuthenticatorAnswerChallengeRequest = AnswerChallengeRequestBuilder.builder()
+                        .withStateHandle(introspectResponse.getStateHandle())
+                        .withCredentials(credentials)
+                        .build();
+
+                IDXResponse resetPasswordResponse = remediationOption.proceed(client, passwordAuthenticatorAnswerChallengeRequest);
+
+                if (resetPasswordResponse.isLoginSuccessful()) {
+                    // login successful
+                    logger.info("Login Successful!");
+                    tokenResponse = resetPasswordResponse.getSuccessWithInteractionCode().exchangeCode(client, idxClientContext);
+                    authenticationResponse.setAuthenticationStatus(AuthenticationStatus.SUCCESS);
+                    authenticationResponse.setTokenResponse(tokenResponse);
+                    return authenticationResponse;
+                } else {
+                    logger.error("Unexpected remediation {}", RemediationType.SUCCESS_WITH_INTERACTION_CODE);
+                }
+            }
+        } catch (ProcessingException e) {
```

Comment:
What about exceptions that happens outside of ION world? For example, when you first hit `interact`; is it also a `ProcessingException` ?

Reply:
Yes.